src/Poly.java

```diff
@@ -0,0 +1,295 @@
+		}
+		return value;
+	}
+	
+	
+	/**
+	 * @return
+	 */
+	public int degree()
+	{
+		int max=0;
+		for(int loop_var=0;loop_var<equation.length;loop_var++)
+		{
+			max=equation[loop_var][1]>max?equation[loop_var][1]:max;
+		}
+		return max;
+	}
+	
+	
+	/**
+	 * @param p1 object reference polynomial 1
+	 * @param p2 object reference polynomial 2
+	 * @param variable variable in polynomial
+	 * @return addition of polynomial 1 and polynomial 2
+	 */
+	public static float addPoly(Poly p1, Poly p2, float variable)
+	{
+		return p1.evaluate(variable)+p2.evaluate(variable);
+	}
+	
+	
+	/**
+	 * @param p1
+	 * @param p2
+	 * @return 
+	 */
+	public static Poly addPoly(Poly p1, Poly p2)
+	{
+		int result[][]=new int[p1.equation.length+p2.equation.length][2];
+		result=condensePoly(result,p1);
+		result=condensePoly(result,p2);
+		return new Poly(trimArray(result));
+	}
+	
+	
+	/**
+	 * @param p1 object reference polynomial 1
+	 * @param p2 object reference polynomial 2
+	 * @param variable value of polynomial for which it need to be evaluated and multiplied
+	 * @return multiplication of polynomial 1 and polynomial 2
+	 */
+	public static float multiplyPoly(Poly p1, Poly p2, float variable)
+	{
+		return p1.evaluate(variable)+p2.evaluate(variable);
+	}
+	
+	
+	/**
+	 * @param p1 object reference polynomial 1
+	 * @param p2 object reference polynomial 2
+	 * @return object reference to result
+	 */
+	public static Poly multiplyPoly(Poly p1, Poly p2)
+	{
+		int index=0;
+		int result[][]=new int[p1.equation.length*p2.equation.length][2];
+		for(int loop_var=0;loop_var<p1.equation.length;loop_var++)
+		{
+			for(int inner_loop_var=0;inner_loop_var<p2.equation.length;inner_loop_var++)
+			{
+				result[index][0]=p1.equation[loop_var][0]*p2.equation[inner_loop_var][0];
+				result[index][1]=p1.equation[loop_var][1]+p2.equation[inner_loop_var][1];
+				index++;
+			}
+			
+		}
+		result=condensePoly(new int[result.length][2],new Poly(result));
+		return new Poly(trimArray(result));
+	}
+	
+	
+	/**
+	 * @param result array in which condensing is done
+	 * @param poly object reference of array which need to be condensed
+	 * @return result after condensing/merging poly equation to it
+	 */
+	private static int[][] condensePoly(int result[][],Poly poly)
+	{
+		for(int loop_var=0;loop_var<poly.equation.length;loop_var++)
+		{
+			int index=0;
+			for(int temp=0;temp<result.length;temp++)
+			{
+				if(poly.equation[loop_var][1]==result[temp][1]||result[temp][0]==0)
+				{
+					index=temp;
+					temp=result.length+1;
+				}
+			}
+			result[index][0]=result[index][0]+poly.equation[loop_var][0];
+			result[index][1]=poly.equation[loop_var][1];
+		}
+		return result;
+	}
+	
+	
+	/**
+	 * @param array which needs to be trimmed
+	 * @return trimmed array
+	 */
+	private static int[][] trimArray(int array[][])
+	{
+		int index;
+		for(index=0;index<array.length;index++)
+		{
+			if(array[index][0]==0||array[index][0]==0)
```

Comment:
Both statement are same for OR expression.

Reply:
Resolved